packages/swirl-components/src/components/swirl-popover-trigger/swirl-popover-trigger.tsx

```diff
@@ -72,9 +78,34 @@ export class SwirlPopoverTrigger {
     }
   }
 
+  private setupHoverListeners() {
+    const popoverEl = this.getPopoverEl();
+
+    popoverEl.addEventListener("mouseenter", () => {
+      if (this.triggerIsActive) {
+        this.stopHoverLingerTimer();
+      }
+    });
+    popoverEl.addEventListener("mouseleave", () => {
+      if (this.triggerIsActive) {
+        this.mouseleaveHandler();
+      }
+    });
```

Comment:
Just one more clean up step: Web components can implement a `disconnectedCallback()` method to do stuff like remove event handlers when the component is removed from the DOM. These two listeners should be removed there. Otherwise the listeners on the popover stay active even if the trigger has been removed. That would cause errors.

Reply:
My understanding is that you cannot remove event handlers that have a reference to the component because they use the arrow functions. Do you know of a way around this?

https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#the_value_of_this_within_the_handler

Reply:
You can make the callbacks a class member and use that reference to add and remove the listener. E.g.:

```ts
popoverOnMouseenter = () => {
  if (this.triggerIsActive) {
    this.stopHoverLingerTimer();
  }
}
```

Reply:
yeah that's the problem, that doesn't work because the context for 'this' is the element that triggered the event, not the class that defined the callback. very weird javascript stuff.

I made it work now after trial and error by passing in the 'this' context through binding it to the component..

Reply:
Using an arrow function for the class member like in my example should work.

`methodName = () => {}` instead of `methodName() {}`

`.bind(this)` works as well, of course.